icu4c/source/common/loclikely.cpp

```diff
@@ -1193,15 +1236,23 @@ uloc_addLikelySubtags(const char* localeID,
     return reslen;
 }
 
-U_CAPI void U_EXPORT2
-ulocimp_addLikelySubtags(const char* localeID,
-                         icu::ByteSink& sink,
-                         UErrorCode* status) {
+static UBool
```

Comment:
See previous comment about introducing parallel error reporting paths.

Reply:
The so called "parallel error reporting paths" pre-exist in createLikelySubtagsString because the API treatment of "no match" in addLikelySubtags and the algorithm for minimizeSubtags calling addLikelySubtags are different. As you can see the current code in _uloc_addLikelySubtags ALSO depend on the return UBool of createLikelySubtagsString() to perform different action and cannot depend on the error code of createLikelySubtagsString because it won't be able to figure out the differences between the  "no match" error from others.

Reply:
So this returns FALSE when there was an error, or no error and no data. And the caller copies the input locale if this function didn't actually add likely subtags. Seems reasonable to me.